### PR TITLE
Added WIP Geo Layout Exporter and some code cleanups

### PR DESCRIPTION
**Pull request by KiritoDv**

### CODE REVIEW

File: src/Companion.h [hunk 1]
```diff
@@ -24,48 +24,77 @@ enum class GBIVersion {
...
+    GBIVersion version = GBIVersion::f3d;
+    GBIMinorVersion subversion = GBIMinorVersion::None;
+};
+
+struct TorchConfig {
+    GBIConfig gbi;
+    SegmentConfig segment;
+    std::string outputPath;
+    ExportType exporterType;
+    bool otrMode;
```
Comment (MegaMech) [inline src/Companion.h:46]: Consider `uint32_t mode` and use an enum with otr, decomp, etc.

File: src/factories/BaseFactory.h [hunk 2]
```diff
@@ -27,6 +28,33 @@ enum class ExportType {
...
+        return std::nullopt;
+    }
+
+    return std::optional<T>(node[key].as<T>());
+}
+
+template<typename T>
+T GetSafeNode(YAML::Node& node, const std::string& key) {
+    if(!node[key]) {
+        throw std::runtime_error("Failed to find " + key);
```
Comment (MegaMech) [inline src/factories/BaseFactory.h:43]: "Failed to find " + key + "in yaml"

File: src/factories/DisplayListOverrides.h [hunk 3]
```diff
@@ -3,6 +3,36 @@
...
+    uint32_t w1;
+} Words;
+
+typedef union {
+    Words words;
+    long long int	force_structure_alignment;
+} Gfx;
+
+typedef struct {
+    short		    ob[3];
```
Comment (MegaMech) [inline src/factories/DisplayListOverrides.h:17]: shouldn't these be int16_t ? or whatever?

**Review by MegaMech — CHANGES_REQUESTED** [review]

File: src/factories/DisplayListOverrides.h (hunk 3 above)
Comment (KiritoDv) [reply src/factories/DisplayListOverrides.h:17]: Yeah but both works, in the end they are the same

**Pull request merged**